File: src/core/ImageType.ts
import IntTypes from './IntTypes.js'
import PixelTypes from './PixelTypes.js'

class ImageType {
  constructor(
    public readonly dimension: number = 2,
    public readonly componentType: typeof IntTypes[keyof typeof IntTypes] = IntTypes.UInt8,
    public readonly pixelType: typeof PixelTypes[keyof typeof PixelTypes] = PixelTypes.Scalar,
    public readonly components: number = 1
  ) {}
}

export default ImageType

File: packages/core/java/src/main/java/org/itk/wasm/ImageType.java
package org.itk.wasm;

public class ImageType {
	public int dimension;
	public IntTypes componentType;
	public PixelType pixelType;
	public int components;

	public ImageType() {
		this.dimension = 2;
		this.componentType = IntTypes.UInt8;
		this.pixelType = PixelType.Scalar;
		this.components = 1;
	}

	// Add getters and setters for the fields
}
